src/components/RepoBranchesDrawer/RepoBranchesDrawer.tsx

```diff
@@ -1,50 +1,44 @@
-import React, { useState } from 'react';
-
-import GitHubStore from '@store/GitHubStore';
-import { BranchItem } from '@store/GitHubStore/types';
+import React from 'react';
+
+import ErrorComponent from '@components/ErrorComponent';
+import Loader from '@components/Loader';
+import { useReposContext } from '@pages/ReposSearchPage/ReposSearchPage';
+import RepoBranchesStore from '@store/RepoBranchesStore';
+import { Meta } from '@utils/meta';
+import { useLocalStore } from '@utils/useLocalStore';
 import { Drawer } from 'antd';
+import { observer } from 'mobx-react-lite';
 import { useParams } from 'react-router-dom';
 
 export type RepoBranchesDrawerProps = {
-  organization: string;
   onClose: () => void;
-  gitHubStore: GitHubStore;
 };
 
-const RepoBranchesDrawer: React.FC<RepoBranchesDrawerProps> = ({
-  organization,
-  onClose,
-  gitHubStore,
-}) => {
+const RepoBranchesDrawer: React.FC<RepoBranchesDrawerProps> = ({ onClose }) => {
   const { name } = useParams<{ name: string }>();
-  const [branchesList, setBranchesList] = useState<BranchItem[]>([]);
-
-  const getRepoBranches = async () => {
-    if (!name) return;
-
-    const result = await gitHubStore.getBranchesList({
-      owner: organization,
-      repo: name,
-    });
-    result.success ? setBranchesList(result.data) : setBranchesList([]);
-  };
+  const repoBranchesStore = useLocalStore(() => new RepoBranchesStore());
```

Comment:
можно сразу инициализовать стор с названием организации и название репозитория
 ```suggestion
  const repoBranchesStore = useLocalStore(() => new RepoBranchesStore(name, reposContext.reposListStore.orgName));
```
 Далее при открытии просто вызывать repoBranchesStore.getRepoBranches().
Так как по логике у тебя стор отвечает за один репозиторий.